Sources/PT.PM.UstPreprocessing/UstPreprocessor.cs

```diff
@@ -146,6 +146,43 @@ public override UstNode Visit(IfElseStatement ifElseStatement)
             return result;
         }
 
+        public override UstNode Visit(UnaryOperatorExpression unaryOperatorExpression)
+        {
+            UstNode result;
+            UnaryOperatorLiteral op = unaryOperatorExpression.Operator;
+            Expression ex = unaryOperatorExpression.Expression;
+            bool isOperatorMinus = op.UnaryOperator == UnaryOperator.Minus;
+            bool isExpressionInt = ex.NodeType == NodeType.IntLiteral;
+            bool isExpressionFloat = ex.NodeType == NodeType.FloatLiteral;
+            if (isOperatorMinus && isExpressionInt)
+            {
+                long intValue = ((IntLiteral)ex).Value;
+                result = new IntLiteral
+                {
+                    Value = -intValue,
+                    FileNode = unaryOperatorExpression.FileNode,
+                    TextSpan = op.TextSpan.Union(ex.TextSpan)
+                };
+                Logger.LogDebug($"Unary expression {unaryOperatorExpression} has been folded to {-intValue} at {result.TextSpan}");
+            }
+            else if(isOperatorMinus && isExpressionFloat)
```

Comment:
Missing space between `if` and `(isOperatorMinus`. `if (isOperatorMinus)` statement can be checked one time:
```CSharp
if (isOperatorMinus)
{
    if (isExpressionInt)
        ....
    if (isExpressionFloat)
        ....
}
else
{
    ....
}
```